### PR TITLE
Add TOKEN_AUDIENCES for configuring audiences in the token review request

### PR DESCRIPTION
**Pull request by yangminzhu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: security/pkg/k8s/tokenreview/k8sauthn.go [hunk 1]
```diff
@@ -24,12 +24,14 @@ import (
...
+	"istio.io/pkg/env"
 )
 
-const (
-	// The default audience for SDS trustworthy JWT. This is to make sure that the CSR requests
+var (
+	// DefaultAudience specifies a list of audiences for SDS trustworthy JWT. This is to make sure that the CSR requests
 	// contain the JWTs intended for Citadel.
-	DefaultAudience = "istio-ca"
+	DefaultAudience = env.RegisterStringVar("TOKEN_AUDIENCES", "istio-ca",
```
Comment (howardjohn) [inline security/pkg/k8s/tokenreview/k8sauthn.go:33]: nit: add strings.Split(...Get()) here so we don't need to do it every time we call it

Comment (yangminzhu) [reply security/pkg/k8s/tokenreview/k8sauthn.go]: done, also updated the variable name and comments.

**Pull request merged**